quickstart/README.adoc

```diff
@@ -50,30 +400,321 @@ To retrieve fruits:
+                "datacenter": "datacenter1",
+                "numberOfNodes": 1
+            }
+        }
+    ]
+}
+----
+
+TIP: If you need health checks globally enabled in your application, but don't want to activate
+Cassandra health checks, you can disable Cassandra health checks by setting the
+`quarkus.cassandra.health.enabled` property to `false` in your `application.properties`.
+
+== Metrics
+
+The Cassandra Quarkus client can provide metrics about the Cassandra session and about individual
+Cassandra nodes.
+
+This behavior is opt-in; it must first be enabled by setting the `quarkus.cassandra.metrics.enabled`
+property to `true` in your `application.properties`.
+
+Then you will also need to add two additional dependencies:
+
+1. `io.quarkus:quarkus-smallrye-metrics`, which enables metrics globally; and
+2. `com.datastax.oss:java-driver-metrics-microprofile`, which enables Cassandra metrics.
+
+These dependencies are already included in this guide's pom.xml.
+
+The next step that you need to do is to define which Cassandra metrics should be enabled. The
+`quarkus.cassandra.metrics.session.enabled` and `quarkus.cassandra.metrics.node.enabled` properties
+should be used for enabling metrics; the former should contain a list of session-level metrics to
+enable, while the latter should contain a list of node-level metrics to enable. Both properties
+accept a comma-separated list of valid metric names.
+
+For example, let's assume that you wish to enable the following three Cassandra metrics:
+
+* Session-leel metrics: `session.connected-nodes` and `session.bytes-sent`;
+* Node-level metrics: `node.pool.open-connections`.
+
+Then you should add the following settings to your `application.properties`:
+
+[source, properties]
+----
+quarkus.cassandra.metrics.enabled=true
+quarkus.cassandra.metrics.session.enabled=connected-nodes,bytes-sent
+quarkus.cassandra.metrics.node.enabled=pool.open-connections
+----
+
+This guide's application.properties file has already many metrics enabled; you can use its metrics
+list as a good starting point for exposing useful Cassandra metrics in your application.
+
+TIP: For the full list of available metric names, please refer to the
+link:https://docs.datastax.com/en/developer/java-driver/latest/manual/core/configuration/reference/[driver
+settings reference] page; search for the `advanced.metrics` section. Also, Cassandra driver metrics
+are explained in detail in the
+https://docs.datastax.com/en/developer/java-driver/latest/manual/core/metrics/[driver manual].
+
+When metrics are properly enabled, metric reports for all enabled metrics are available at the
+`/metrics` REST endpoint of your application.
+
+Running in dev mode with `mvn clean quarkus:dev`, if you point your browser to
+http://localhost:8080/metrics you should see a list of metrics; search for metrics whose names start
+with `vendor_cassandra`.
+
 == Running in native mode
 
 You can compile the application into a native binary using:
```

Comment:
This was my first time ever using GraalVM and it's native compilation, I didn't expect the time that it took to complete the native build.  Might be worth mentioning here that native compilation can take a significant amount of time.

I was running inside of a lower-end VM though, so potentially it's faster on a higher end machine.